monai/transforms/croppad/dictionary.py

```diff
@@ -268,11 +313,31 @@ def __init__(
     def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.ndarray]:
         d = dict(data)
         for key in self.key_iterator(d):
+            self.push_transform(d, key)
             d[key] = self.cropper(d[key])
         return d
 
+    def inverse(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.ndarray]:
+        d = deepcopy(dict(data))
+
+        for key in self.key_iterator(d):
+            transform = self.get_most_recent_transform(d, key)
+            # Create inverse transform
+            orig_size = transform[InverseKeys.ORIG_SIZE.value]
+            pad_to_start = np.array(self.cropper.roi_start)
+            pad_to_end = orig_size - self.cropper.roi_end
```

Comment:
the forward part of this has some tolerances for too large cropping values  https://github.com/Project-MONAI/MONAI/blob/master/monai/transforms/croppad/array.py#L239  (taking max over the values). Do we care about this kind of details in this PR? I feel it's gonna be lots of efforts

Reply:
I don't think we need to take care of these, since they're taken care of in the forward direction. I.e., in line 239 of the link you sent, `self.roi_end` is updated. This means that the amount of cropping is performed with `self.roi_end` and then the amount of padding in the inverse uses the same value.

Reply:
I was trying to add this use case to `test_inverse`:
```py
TESTS.append(
    (
        "SpatialCropd 2d",
        "2D",
        0,
        SpatialCropd(KEYS, [49, 51], [390, 89]),
    )
)
```
the forward works fine, but in the inverse:
```
Traceback (most recent call last):
  File "/usr/local/anaconda3/envs/py37/lib/python3.7/site-packages/parameterized/parameterized.py", line 533, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/Documents/MONAI/tests/test_inverse.py", line 301, in test_inverse
    fwd_bck = t.inverse(fwd_bck)
  File "/Documents/MONAI/monai/transforms/croppad/dictionary.py", line 333, in inverse
    d[key] = inverse_transform(d[key])
  File "/Documents/MONAI/monai/transforms/croppad/array.py", line 152, in __call__
    raise ValueError(f"self.spatial_border must contain only nonnegative ints, got {spatial_border}.")
ValueError: self.spatial_border must contain only nonnegative ints, got (0, -289, 7, 4).

```

Reply:
In [this commit](https://github.com/Project-MONAI/MONAI/pull/1737/commits/25c5daad158d4d7e3398f286d18c153c7d31b7f4), I addressed your concern by modifying the forward pass of `BorderPadd`. -ve padding values are set to 0. What do you think? Should be backwards compatible since previously it would raise an exception.

Reply:
I've also added extra edge cases for the other croppad. Modifying the forward of `BorderPadd` allows for little changes to the inverse methods.